pkg/providers/v2/cloud.go

```diff
@@ -19,14 +19,21 @@ limitations under the License.
 package v2
 
 import (
+	"fmt"
 	"io"
+	"regexp"
 
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/credentials/ec2rolecreds"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/session"
 	cloudprovider "k8s.io/cloud-provider"
 )
 
 func init() {
 	cloudprovider.RegisterCloudProvider(ProviderName, func(config io.Reader) (cloudprovider.Interface, error) {
-		return &cloud{}, nil
+		return newCloud()
```

Comment:
@andrewsykim updated! 
Also passed the unit test, let me know if you have any other suggestions :)